Rename Testcontainers for C to Testcontainers Native

seffradev opened the pull request:
<!-- Type of change
Please label this PR with one of the existing labels, depending on the scope of your change.
-->

## What does this PR do?

This renames every instance of "Testcontainers for C", "Testcontainers C", and "testcontainers-c".

<!-- Mandatory
Explain here the changes you made on the PR. Please explain the WHAT: patterns used, algorithms implemented, design architecture, etc.
-->

## Why is it important?

It makes the naming consistent throughout the project. I am expecting a minimal breakage due to renamed headers and project, but should be a quick fix.

<!-- Mandatory
Explain here the WHY, or the rationale / motivation for the changes.
-->

## Related issues

<!-- Recommended
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.

- Closes #123
- Relates #123
- Requires #123
- Supersedes #123
-->
- None.

<!-- Recommended
## How to test this PR

Explain here how this PR will be tested by the reviewer: commands, dependencies, steps, etc.
-->

<!-- Optional
## Follow-ups

Add here any thought that you consider could be identified as an actionable step once this PR is merged.
-->

oleg-nenashev commented:
@seffradev, should we rename all the instances? Or should we keep the naming for distributables, e.g.?

* `Testcontainers Native` - the repo, site, project name, etc.
* `Testcontainers for C` - a low-level library for C, and the respective packages
* `Testcontainers for C++` - a header lib for C++, and the respective packages
* ... - whatever else gets added

seffradev commented:
It is a fair question, and I will elaborate (and possibly change this PR) when I've had time to write a thorough proposal for a design that addresses your concerns on the C API design.

In short, I'm thinking:
- A Go-to-C bridge (called `testcontainers-native-bridge` or something) that is supposed to be internal.
- A C library, consuming `native-bridge` (and this would be called `testcontainers-c`), which allows us to write an idiomatic C library that abstracts all Go code from library consumers.
- After that, a C++ library adapter, called `testcontainers-cpp`.
- And possibly other language wrappers after that.

The proposal will also mention ideas for how the modules (currently WireMock) would be set up and designed around this.

oleg-nenashev commented:
@seffradev fine be me, and thanks. I think we are on the same page.
Definitely "Testcontainers Naative" can remain the umbrella name for the project.
No hurry to rename w.r.t the proposal

seffradev commented:
I'm closing this in favor of #53.

oleg-nenashev commented:
OK, I think we agree in principle. I will suggest a patch with naming conventions to CONTRIBUTING, and then we can align the naming
